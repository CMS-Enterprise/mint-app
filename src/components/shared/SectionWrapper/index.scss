.easi-section {
    &__border {
        border: 1px solid color($theme-color-base-light);
    }
    &__border-bottom {
        border-bottom: 1px solid color($theme-color-base-light);
    }
    &__border-top {
        border-top: 1px solid color($theme-color-base-light);
    }
}
