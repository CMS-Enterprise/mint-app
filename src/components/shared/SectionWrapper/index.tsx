import React, { ReactNode } from 'react';
import classnames from 'classnames';

import './index.scss';

type SectionWrapperProps = {
  className?: string;
  children?: ReactNode;
  border?: boolean;
  borderBottom?: boolean;
  borderTop?: boolean;
};

const SectionWrapper = ({
  className,
  children,
  border,
  borderBottom,
  borderTop
}: SectionWrapperProps) => {
  const classNames = classnames(
    'easi-section',
    {
      'easi-section__border': border,
      'easi-section__border-bottom': borderBottom,
      'easi-section__border-top': borderTop
    },
    className
  );
  return (
    <div data-testid="section-wrapper" className={classNames}>
      {children}
    </div>
  );
};

export default SectionWrapper;
