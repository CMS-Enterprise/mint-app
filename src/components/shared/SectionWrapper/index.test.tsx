import React from 'react';
import { render } from '@testing-library/react';

import SectionWrapper from './index';

describe('SectionWrapper', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<SectionWrapper />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with a border', () => {
    const { getByTestId } = render(<SectionWrapper border />);

    expect(getByTestId('section-wrapper')).toHaveClass('easi-section__border');
  });

  it('renders with a bottom border', () => {
    const { getByTestId } = render(<SectionWrapper borderBottom />);

    expect(getByTestId('section-wrapper')).toHaveClass(
      'easi-section__border-bottom'
    );
  });

  it('renders with a top border', () => {
    const { getByTestId } = render(<SectionWrapper borderTop />);

    expect(getByTestId('section-wrapper')).toHaveClass(
      'easi-section__border-top'
    );
  });
});
