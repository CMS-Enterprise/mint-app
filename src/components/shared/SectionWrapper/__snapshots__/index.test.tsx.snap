// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionWrapper matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="easi-section"
    data-testid="section-wrapper"
  />
</DocumentFragment>
`;
